Collections/script.js

```diff
@@ -253,17 +256,161 @@ console.log(filterByAge(people, 30));
+let objWords = {}; // deklaruję nowy pusty objekt
+arrWords.forEach( (element) => {  // dla każdego elementu nowej  tablicy arrWords tworzę funkcję, która :
+    let sum = arrWords.filter((word) => word === element); // filtruje z tej tablicy wszystkie wartości które są równe wartości elementu na którym obecnie występuje iteracja. Przypisuję te wartości do nowej zmiennej "sum" dzięki temu mam tablicę zawierającą tylko te elementy ( i tyle elementów), które aktualnie iteruję
+    objWords[element] = sum.length; // przypisuję do nowego objektu objWords wartość, którą jest jako klucz objektu nazwa elementu iterowanego a jako wartość klucza długość aktualnej tablicy sum. 
+   
+})
+
+return objWords // zwracam cały objekt 
+}
+
+console.log(countWordOccurrences("JavaScript byl dla mnie skomplikowany a teraz JavaScript jest dla mnie znacznie prostszy. Viva Javascript"))
+
+
+*/
+// // Zadanie 16.
+/*
+
+
+function toCamelCase(string) {  // deklaruję funkcję przyjmującą jako parametr  typ String
+    let arrWords = string.replaceAll("_", " ") // deklaruję zmienną, której wartością jest String użyty jako argument funkcji z usuniętym znakiem "_" ( zamieniony na pusty znak metodą replace)     
+    arrWords = arrWords.replaceAll("-", " ") // jak wyżej tylko zamieniam dodatokwo znak "-" na w.w. zmiennej, która już jest pozbawiona znaków "_" ( pewnie da się to jakoś jedną metodą zrobić - zamienić dwa różne znaki na pusty string = ale ja nie wiem jak więc wykonuję w dwóch etapach)
+    arrWords = arrWords.split(" ") // splituję wartość zmiennej i uzyskuje tablicę z poszczególnymi wyrazami jako elementy ( już bez znaków "-" oraz "_")
+for ( let i = 1; i < arrWords.length; i++ ) {  // iteruję tablicę zaczynając od drugiego elementu czyli index 1. Pierwszy chcemy aby pozostawał bez zmian.
+    arrWords[i] = arrWords[i].replace(arrWords[i][0], arrWords[i][0].toUpperCase() ) // zamieniam drugi a następnie każdy kolejny element tablicy w taki sposób, że dla każdego takiego elementu pobieram za pomocą metody replace  pierwszą wartość tego elementu ( index 0 elementu i) i zamieniam go na powiększony metodą toUpperCase.
+ }
+let camelCase = arrWords.join("")  // deklaruję nową zmienną do której przypisuję połączone elementy tablicy arrWords już z zamienionymi pierwszymi literami.
+  return camelCase // zwracam zmienną i drukuję funkcję z przykładowymi stringami jako argument
+}
+console.log(toCamelCase("java_script")); // => javaScript
+console.log(toCamelCase("java-script")); // => javaScript
+console.log(toCamelCase("Java-Script")); // => JavaScript
+console.log(toCamelCase("asp_Net_Core")); // => aspNetCore
+
+
+*/
+// // Zadanie 17.
+/*
+
+
+function getStudentTopGrades(arr) { // deklaruję funkcję przyjmującą jako parametr tablicę
+let topGrades = arr.map(element => element.grades)  // deklaruję nową tablicę, której elementami będą wszystkie oceny każdego ucznia
```

Comment:
O ile wartości pod tą zmienną nie zamierzasz później nadpisywać, to dobrą praktyką jest używanie `const` nie `let`